### PR TITLE
Update some p6est docs

### PR DESCRIPTION
**Pull request by jkozdon**
In calls to `replace_fn` the number of out layers should be set to `4` when horizontal coarsening. See docs for `p6est_replace_t` in `src/p6est.h`.

### CODE REVIEW

**Comment by jkozdon** [comment]
cc @tisaac

**Comment by jkozdon** [comment]
Not sure if this is the right branch or not for this fix.

File: src/p6est.c [hunk 1]
```diff
@@ -1636,7 +1636,7 @@ p6est_replace_column_join (p4est_t * p4est, p4est_topidx_t which_tree,
       }
     }
     if (replace_fn != NULL) {
-      replace_fn (p6est, which_tree, P4EST_CHILDREN, 1, outgoing, q, 1, 1,
+      replace_fn (p6est, which_tree, P4EST_CHILDREN, 4, outgoing, q, 1, 1,
```
Comment (cburstedde) [inline src/p6est.c:1639]: Good point, but having Toby confirm may still be the safest. Let's make this P4EST_CHILDREN for clarity.
Testing the change, does everything check out, i.e., the pointer to the outlayers contains 4 of them and they are valid?
While you're at it, if you have anything to add to the documentation of the replace function type please feel free.

Comment (jkozdon) [reply src/p6est.c]: Done.

**Comment by jkozdon** [comment]
> Testing the change, does everything check out, i.e., the pointer to the outlayers contains 4 of them and they are valid?

Based on the code above the correction, `q` is of size `P4EST_CHILDREN` and all the entries are initialized.

- https://github.com/cburstedde/p4est/blob/prev3-develop/src/p6est.c#L1574
- https://github.com/cburstedde/p4est/blob/prev3-develop/src/p6est.c#L1601
- https://github.com/cburstedde/p4est/blob/prev3-develop/src/p6est.c#L1632

**Comment by jkozdon** [comment]
closes #128

**Comment by tisaac** [comment]
I'm sorry I'm late getting this!

This change seems wrong to me, but it's a bugfix so there must be something I don't understand.

At the call site `replace_fn` should be used to update the data for a change of the following kind:

```
+-----+-----+           +-----------+    
|\     \     \          |\           \
| +-----\-----+         | \           \
+ |\     \     \   =>   +  \           \
 \| +-----+-----+        \  +-----------+
  + |     |     |         \ |           |
   \|     |     |          \|           |
    +-----+-----+           +-----------+
```

The description is confusing and should be updated because "outgoing" means leaving the mesh (left) and "incoming" means new and joining the mesh (right), but controlling for that the number of new layers is 1, not 4.

**Comment by jkozdon** [comment]
> This change seems wrong to me, but it's a bugfix so there must be something I don't understand.

It's also very possible I have misunderstood the expected behavior.

**Comment by tisaac** [comment]
Assuming you have data attached to the layers in the columns, `replace_fn` should be used to initialize data for incoming layers from the data in the outgoing layers that occupy the same location in the mesh.  I guess my questions are

- Is that what you were expecting?
- What kind of bug did you experience?

**Comment by jkozdon** [comment]
Honestly, I am not sure.

Relooking at this I did a few months ago I am not sure why I thought that outlayers should be 4.

I've removed that commit. The comment fixes might still be useful?

**Comment by cburstedde** [comment]
Happy to merge if @tisaac signs off. Thanks for keeping at it!

**Pull request merged**